### PR TITLE
use self hosted runner and disable windows workflow

### PR DESCRIPTION
**Pull request by ruanyl**
### Description

Use self hosted runner and disable windows workflow + use forked functional-test repo for development

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: .github/workflows/build_and_test_workflow.yml [hunk 1]
```diff
@@ -307,6 +298,7 @@ jobs:
           retention-days: 1
 
   bwc-tests:
+    if: false
```
Comment (SuZhou-Joe) [inline .github/workflows/build_and_test_workflow.yml:301]: Maybe we can enable bwc-tests?

**Review by Hailong-am — APPROVED** [review]

File: .github/workflows/cypress_workflow.yml [hunk 2]
```diff
@@ -54,7 +54,7 @@ jobs:
         uses: actions/checkout@v2
         with:
           path: ${{ env.FTR_PATH }}
-          repository: opensearch-project/opensearch-dashboards-functional-test
+          repository: ruanyl/opensearch-dashboards-functional-test
           ref: '${{ github.base_ref }}'
```
Comment (raintygao) [inline .github/workflows/cypress_workflow.yml:58]: @ruanyl @SuZhou-Joe Do we need to set ref to `workspace` branch of ft repo to do workspace-specific test integration?

Comment (SuZhou-Joe) [reply .github/workflows/cypress_workflow.yml:58]: Both should work, what about using workspace branch as the ref for our development? @ruanyl

Comment (raintygao) [reply .github/workflows/cypress_workflow.yml:58]: There have been some commits to fix workspace failed FT on the workspace branch. https://github.com/opensearch-project/opensearch-dashboards-functional-test/commits/workspace

**Review by raintygao — APPROVED** [review]

**Pull request merged**